src/sagas/task.saga.js

```diff
@@ -0,0 +1,339 @@
+      }
+    }
+  } finally {
+    /* child process has completed */
+  }
+}
+
+export function* taskAbort({ task }: { task: Task }): Saga<void> {
+  const { processId, name } = task;
+
+  yield call(killProcessId, processId);
+  yield call([ipcRenderer, ipcRenderer.send], 'removeProcessId', processId);
+
+  // Once the children are killed, we should dispatch a notification
+  // so that the terminal shows something about this update.
+  // My initial thought was that all tasks would have the same message,
+  // but given that we're treating `start` as its own special thing,
+  // I'm realizing that it should vary depending on the task type.
+  // TODO: Find a better place for this to live.
+  const abortMessage = isDevServerTask(name)
+    ? 'Server stopped'
+    : 'Task aborted';
+
+  yield put(receiveDataFromTaskExecution(task, chalk.bold.red(abortMessage)));
+}
+
+export function* displayTaskComplete(task: Task): Saga<void> {
+  // Send a message to add info to the terminal about the task being done.
+  // TODO: ASCII fish art?
+
+  const message = 'Task completed';
+
+  yield put(receiveDataFromTaskExecution(task, chalk.bold.green(message)));
+}
+
+export function* taskComplete({ task }: { task: Task }): Saga<void> {
+  if (task.processId) {
+    yield call(
+      [ipcRenderer, ipcRenderer.send],
+      'removeProcessId',
+      task.processId
+    );
+  }
+
+  // The `eject` task is special; after running it, its dependencies will
+  // have changed.
+  // TODO: We should really have a `EJECT_PROJECT_COMPLETE` action that does
+  // this instead.
+  if (task.name === 'eject') {
+    const project = yield select(getProjectById, task.projectId);
+
+    yield put(loadDependencyInfoFromDisk(project.id, project.path));
+  }
+}
+
+const createStdioChannel = (
+  child: any,
+  handlers: {
+    stdout: (emitter: any) => (data: string) => void,
+    stderr: (emitter: any) => (data: string) => void,
+    exit: (emitter: any) => (code: number) => void,
+  }
+) => {
+  return eventChannel(emitter => {
+    child.stdout.on('data', handlers.stdout(emitter));
+    child.stderr.on('data', handlers.stderr(emitter));
+    child.on('exit', handlers.exit(emitter));
+
+    return () => {
+      /* unsubscribe any listeners */
+    };
+
+    // use an expanding buffer because we don't want to lose any information
+    // passed up by the child process. initialize it at a length of 2 because
+    // at bare minimum we expect to have 2 messages queued at some point (as
+    // the exit channel completes, it should emit the return code of the process
+    // and then immediately END
+  }, buffers.expanding(2));
+};
+
+export const getBaseProjectEnvironment = (projectPath: string) => ({
+  // Forward the host env, and append the
+  // project's .bin directory to PATH to allow
+  // package scripts to function properly.
+  ...window.process.env,
+  PATH:
+    window.process.env.PATH +
+    path.delimiter +
+    path.join(projectPath, 'node_modules', '.bin'),
+});
```

Comment:
Feels like this method should move somewhere... maybe `platform.service.js`?

Reply:
Yeah that definitely makes more sense.